Sources/Build/BuildPlan.swift

```diff
@@ -702,19 +702,22 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
             switch dependency {
             // Include all the dependencies of a target.
             case .target(let target, _):
+                if [.macro, .plugin].contains(target.type) {
+                    return []
+                }
                 return target.dependencies.filter { $0.satisfies(self.buildEnvironment) }
 
             // For a product dependency, we only include its content only if we
-            // need to statically link it or if it's a plugin.
+            // need to statically link it.
             case .product(let product, _):
                 guard dependency.satisfies(self.buildEnvironment) else {
                     return []
                 }
 
                 switch product.type {
-                case .library(.automatic), .library(.static), .plugin:
+                case .library(.automatic), .library(.static):
```

Comment:
Is this being removed because it's already being filtered out on the new line 705 above?

Reply:
No, that's separate. Both are being removed since it doesn't seem right to link dependencies of plugins to the products that are using the plugin. 

Right now what happens is that e.g. if a plugin is depending on an executable, we link the code of the executable into any client of the plugin.

Reply:
i think this will read more intuitively if the macro and plugin cases are separated out to their own case statements